docs/core/oo_feature_summary.md

```diff
@@ -0,0 +1,387 @@
+more `features <property_features>`.
+
+## Interfaces
+
+A leading uppercase `I` is used as a naming convention for `interfaces
+<interfaces>`.
+
+    qx.Interface.define("my.cool.IInterface");
+
+## Mixins
+
+Leading uppercase `M` as a naming convention. A `mixin <mixins>` can
+have all the things a class can have, like properties, constructor,
+destructor and members. :
+
+    qx.Mixin.define("my.cool.MMixin");
+
+## Attaching mixins to a class
+
+The `include` key contains either a reference to an single mixin, or an
+array of multiple mixins: :
+
+    qx.Class.define("my.cool.Class",
+    {
+      include : [my.cool.MMixin, my.other.cool.MMixin]
+      ...
+    });
+
+## Attaching mixins to an already defined class
+
+    qx.Class.include(qx.ui.core.Widget, qx.MWidgetExtensions);
+
+## Access
+
+By the following naming convention. Goal is to be as consistent as
+possible. During the build process private members can optionally be
+renamed to random names in order to ensure that they cannot be called
+from outside the class. :
+
+    publicMember
+    _protectedMember
+    __privateMember
+
+## Static classes
+
+Explicit declaration allows for useful checks during development. For
+example. `construct` or `members` are not allowed for such a purely
+static class. :
+
+    qx.Class.define("my.cool.Class", {
+      type : "static"
+    });
+
+## Abstract classes
+
+Declaration allows for useful checks during development and does not
+require explicit code. :
+
+    qx.Class.define("my.cool.Class", {
+      type : "abstract"
+    });
+
+## Singletons
+
+Declaration allows for useful checks during development and does not
+require explicit code. A method `getInstance()` is added to such a
+singleton class. :
+
+    qx.Class.define("my.cool.Class",
+    {
+      type : "singleton",
+      extend :  my.great.SuperClass
+    });
+
+## Immediate access to previously defined members
+
+The closed form of the class definition does not allow immediate access
+to other members, as they are part of the configuration data structure
+themselves. While it is typically not a feature used very often, it
+nonetheless needs to be supported by the new class declaration. Instead
+of some trailing code outside the closed form of the class declaration,
+an optional `defer` method is called after the other parts of the class
+definition have been finished. It allows access to all previously
+declared `statics`, `members` and dynamic `properties`.
+
+<div class="note">
+
+<div class="admonition-title">
+
+Note
+
+</div>
+
+If the feature of accessing previously defined members is not absolutely
+necessary, `defer` **should not be used** in the class definition. It is
+missing some important capabilities compared to the regular members
+definition and it cannot take advantage of many crucial features of the
+build process (documentation, optimization, etc.).
+
+</div>
+
+    qx.Class.define("my.cool.Class",
+    {
+      statics:
+      {
+        driveLetter : "C"

```

Comment:
Documenting what the defer section will do is very useful for later figuring out where the declarations of variables are located. I would encourage developers to add here:
```
drive : null  // set in defer method
```
 and a members section too:
```
members:
{
  whatsTheDrive : null  // set in defer method
}
```